[bugfix] install requirements and fix readme (fix #1, fix #2, fix #3, fix #4)

mitsururike opened the pull request.

Commit:
[bugfix] install requirements and fix readme (fix #1, fix #2, fix #3, fix #4)

trombik commented on README.md:
```diff
@@ -10,15 +10,58 @@ None
+| `jenkins_slave_user` | Jenkins user | `{{ __jenkins_slave_user }}` |
+| `jenkins_slave_group` | Jenkins group | `{{ __jenkins_slave_group }}` |
+| `jenkins_slave_home` | Jenkins home | `{{ __jenkins_slave_home }}` |
+| `jenkins_slave_package` | package name | `{{ __jenkins_slave_package }}` |
```
`jenkins_slave_java_opts` is not documented

trombik replied:
other undocumented variables:

* `jenkins_slave_ansible_vault_key`
* `jenkins_slave_authorized_keys`

when adding a variable, it is always recommend to define the variable in `defaults/main.yml` to prevent undocumented variables.

trombik replied:
also, why does `tasks/main.yml` include `sample_secret.yml`?

trombik replied:
also, in `README`, please replace a placeholder with something meaningful. a simple description is enough like "Configure jenkins slave". That description should be placed in the repository description (see the top page of the repository and you will find "No description, website, or topics provided.").

```
A brief description of the role goes here
```

trombik replied:
Example playbook should be a working example, and the example in the README does not work. to ensure the example always works, just update the example with `tests/serverspec/default.yml`, which is supposed to work.

Commit:
[bugfix] fix README.md and the unit test
- fix undocumented variables
- sample_secret.yml is used for the unit test and rename/move it to tests
- add a unit test to check the content of .ansible_vault_key
